src/clib/lib/enkf/enkf_obs.cpp

```diff
@@ -1017,35 +1006,42 @@ py::handle pybind_alloc(int history_,
 void enkf_obs_load(Cwrap<enkf_obs_type> enkf_obs, const char *config_file,
                    double std_cutoff) {
 
-    conf_class_type *enkf_conf_class = enkf_obs_get_obs_conf_class();
-    conf_instance_type *enkf_conf = conf_instance_alloc_from_file(
-        enkf_conf_class, "enkf_conf", config_file);
+    const std::unique_ptr<conf_class_type, void (*)(conf_class_type *)>
+        enkf_conf_class(enkf_obs_get_obs_conf_class(), conf_class_free);
+    const std::unique_ptr<conf_instance_type, void (*)(conf_instance_type *)>
+        enkf_conf(conf_instance_alloc_from_file(enkf_conf_class.get(),
+                                                "enkf_conf", config_file),
+                  conf_instance_free);
 
-    const char *errors = conf_instance_get_path_error(enkf_conf);
+    const char *errors = conf_instance_get_path_error(enkf_conf.get());
     if (errors) {
-        enkf_obs_reinterpret_DT_FILE(errors);
-        exit(1); // No need to free errors...
+        throw exc::invalid_argument{
+            "The following keywords in your configuration did not resolve to a "
```

Comment:
Can this happen? Is there test coverage for this?

Reply:
I changed the test so that it should be clear what is going on :)